### ISSUE TITLE
PPrint fails to derive a printer for case classes with more than 1 argument list

### ISSUE THREAD

**Issue by stanch** [lihaoyi/upickle-pprint#148]
``` scala
@ case class A(x: List[String], y: Int) 
defined class A
@ A(List.fill(10)("long string"), 0) 
res12: A = A(
  List(
    "long string",
    "long string",
    "long string",
    "long string",
    "long string",
    "long string",
    "long string",
    "long string",
    "long string",
    "long string"
  ),
  0
)
@ case class B(x: List[String])(y: Int) 
defined class B
@ B(List.fill(10)("long string"))(0) 
res14: B = B(List(long string, long string, long string, long string, long string, long string, long string, long string, long string, long string))
```

**Comment by Voltir** [comment]
This also appears to impact upickle:

```scala
scala> case class B(x: List[String])(y: Int)
defined class B

scala> upickle.default.write(B(List.fill(10)("long string"))(0))
<console>:13: error: type mismatch;
 found   : Option[List[String]]
 required: Option[(List[String], Int)]
Error occurred in an application involving default arguments.
       upickle.default.write(B(List.fill(10)("long string"))(0))
```

**Comment by stanch** [comment]
From what I can tell, it’s fixed somewhere in 0.5.0. At least ammonite 0.9.0 no longer has this problem.
